Description: Fix unused library linking
 Make the libcrypto conditional on slpv2security, so that we can avoid a
 Build-Conflicts against libssl-dev. And switch library checks to use the
 more correct AC_SEARCH_LIBS checks instead of AC_CHECK_LIB for all
 libraries.
Author: Guillem Jover <guillem@debian.org>
Origin: vendor
Last-Update: 2014-07-25

---
 configure.in |   16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

--- a/configure.in
+++ b/configure.in
@@ -102,9 +102,6 @@ AC_ARG_ENABLE(slpv2-security,
 slpv2security=$enableval,
 slpv2security=no)
 
-AC_CHECK_LIB(crypto, DSA_sign) 
-AC_CHECK_FUNCS(DSA_verify DSA_size DSA_free SHA1)
-
 dnl Conditional for optional security support in libslp SLP API library
 AM_CONDITIONAL(ENABLE_SLPv2_SECURITY, test "$slpv2security" = "yes")
 if test "$slpv2security" = "yes"; then
@@ -170,12 +167,17 @@ OPENSLP_STRUCT_SA_RESTORER
 dnl ***********************************************************************
 dnl Checks for libraries.
 dnl ***********************************************************************
-AC_CHECK_LIB(resolv, inet_aton)
-AC_CHECK_LIB(socket, main)
-AC_CHECK_LIB(nsl, gethostbyname)
-dnl AC_CHECK_LIB(m, main)
+AC_SEARCH_LIBS(resolv, inet_aton)
+AC_SEARCH_LIBS(socket, main)
+AC_SEARCH_LIBS(nsl, gethostbyname)
+dnl AC_SEARCH_LIBS(m, main)
 dnl AC_CHECK_FUNCS(ceil log10 strncasecmp strcasecmp )
 AC_CHECK_FUNCS(strncasecmp strcasecmp)
+if test "$slpv2security" = "yes"; then
+    AC_SEARCH_LIBS([DSA_sign], [crypto])
+    AC_CHECK_FUNCS([DSA_verify DSA_size DSA_free SHA1])
+fi
+
 dnl AC_OUTPUT( Makefile common/Makefile libslpattr/Makefile libslp/Makefile 
 dnl 	slpd/Makefile slptool/Makefile test/Makefile slp.list )
 AC_CONFIG_FILES([ Makefile ])
